### ISSUE TITLE
Thumbnail not generated on server for non-image type file such as .docx

### ISSUE THREAD

**Issue by j-furby** [blackcity/Backload#57]
Firstly everything works great running http://localhost. However when pushing the solution to the server thumbnails are not generated (see image which shows an Example.docx). The file is uploaded to the server and can be downloaded.

![backloadthumbnail](https://cloud.githubusercontent.com/assets/3984827/9543043/7b1d139e-4d6d-11e5-87af-dae36947b3af.png)

Looking in the _thumbs directory on the server I can see the "Example.docx.png" file, but it has 0 bytes. The only difference I can see is that when deploying to the server the solution is not root and sits in a folder like http://server/solution/upload.view

Tried with the default config file and the result is the same (the only change I make is adding filesRoot to ~/Upload). Viewing the documentation I can see the contentType settings for Pro/Enterprise, but I do not want to change the defaults, I'm happy to leave them as they are. 

Is there any other settings I should be aware of when a solution resides in a sub-folder or could this be another issue? Thanks,

James.

Edit, forgot to mention I'm using Backload version 2.0.0.5 from Nuget.

**Comment by Auersberg** [comment]
So, Image files generate a thumbnail correctly? How about pdf?
Make sure the folder for content type images is on your server: "~/Backload/ContentTypes/". This may cause the problem because the server side component recognizes a none image content type and tries to load a default Image for the type from this folder.

**Comment by j-furby** [comment]
Wow fast response! 

The thumbnails for images are generated properly. Loading a PDF yields the same result (see image). 

![backloadfileexception](https://cloud.githubusercontent.com/assets/3984827/9544198/81f85ee0-4d76-11e5-9311-0bdc5722f3ec.png)

I've double checked the IIS settings (apologies my knowledge is limited here), with the default site pointing to d:\inetpub\wwwroot so I've placed a copy of "Backload/ContentTypes/" there with full permissions for now, however the result is the same. 

For clarity the dev server hosts a number of applications each resides in it's own sub-fodler such as:
d:\inetpub\wwwroot\application1
d:\inetpub\wwwroot\application2
d:\inetpub\wwwroot\application3 
etc...

**Comment by Auersberg** [comment]
Can you provide a trace output like shown here? 
https://github.com/blackcity/Backload/wiki/Example-13
Add the system.Diagnostics element to your web.config, choose a trace listener (e.g text or xml output) and set the trace level to "Verbose" within this element. (Note: If you can debug your application with Visual Studio, you can see the trace output in Visual Studios output window, but with trace level "Information" which provides fewer details than "Verbose".)

I am currently not on a dev machine, but I'll do some tests with the info you provided this weekend.

**Comment by Auersberg** [comment]
One more thing: I think the server side component cannot find the content type Images. I am currently not know why. But, there is a fallback you can use: 
- Make a copy of the "~Backload/ContentType" folder in "~/Backload/Client/"
- Rename the new "~/Backload/Client/ContentType" folder to "~/Backload/Client/content-types"
- Upload

This may works, because this folder is a fallback for previous versions.  It will not display a Word thumbnail, but a default thumbnail.

**Comment by j-furby** [comment]
Creating the following D:\inetpub\wwwroot\my_solution\Backload\Client\content-types has resulted in a default thumbnail :)

![content-types](https://cloud.githubusercontent.com/assets/3984827/9546102/d424cd6e-4d87-11e5-8f57-be1fc4d8a344.png)

I've added the following to the Web.Config 

```xml
<system.diagnostics>
    <trace autoflush="true" />
    <sources>
      <source name="Backload" switchName="traceLevel">
        <listeners>
          <add name="logfile" />
          <add name="xmlfile" />
        </listeners>
      </source>
    </sources>
    <sharedListeners>
      <add name="logfile" type="System.Diagnostics.TextWriterTraceListener" initializeData="logs/backload.log" traceOutputOptions="DateTime" />
      <add name="xmlfile" type="System.Diagnostics.XmlWriterTraceListener" initializeData="logs/backload.svclog" traceOutputOptions="DateTime" />
    </sharedListeners>
    <switches>
      <add name="traceLevel" value="Verbose, ActivityTracing" />
    </switches>
  </system.diagnostics>
```

When running locally I can see the verbose output in Visual Studio (all working well), however on the server I cannot find any trace of the logging. Inspecting at the IIS log settings the folder is set to D:\inetpub\logs\LogFiles with format W3C. Viewing this directory show 3 folders with lots of text files, however none of them are recently modified. I will have to speak with some one more knowledgeable about the server setup to dig further.

**Comment by tridip-bba** [comment]
HI,

 

Anyone tell me what is the use of Backload ?

 

What kind of purpose this library solved. Please discuss with scenario for which Backload is used in ASP.Net MVC

 

Thanks

Tridip

**Comment by Auersberg** [comment]
\<add name="logfile" initializeData="logs/backload.log" ... />

Cannot test is at the moment, but "logs" points to "~/Backload/logs" which does not exist. Create a logs folder in your applications root folder (or another place, edit initializeData="" in this case) and give this folder appropriate permissions (you may point it to your Upload folder, as it has set the permissions already)

**Comment by j-furby** [comment]
Hard coding the initializeData path to the upload folder did the trick for the server Web.Config did the trick. However the log does not contain any useful information:

```
Backload Information: 0 : Backload component (2.0.1.0.3373): Static initialization process started.
    DateTime=2015-08-28T12:54:32.7488298Z
Backload Information: 5 : Backload component (2.0.1.0.3373): Static initialization process finished with success.
    DateTime=2015-08-28T12:54:32.7644298Z
Backload Information: 10 : Backload component ({0}): Instance initialization started.
    DateTime=2015-08-28T12:54:32.7956298Z
Backload Information: 0 : Backload component (2.0.1.0.3373): Static initialization process started.
    DateTime=2015-08-28T12:59:45.7940298Z
Backload Information: 5 : Backload component (2.0.1.0.3373): Static initialization process finished with success.
    DateTime=2015-08-28T12:59:45.8096298Z
Backload Information: 10 : Backload component ({0}): Instance initialization started.
    DateTime=2015-08-28T12:59:45.8096298Z
```

**Comment by Auersberg** [comment]
Never mind. Maybe it has not flushed the trace before the component has finished. We now know where to look if there is an error. 
If you want, you can set the path to the content type thumbnail in your code. 

- Start with the events demo in the demo package or here https://github.com/blackcity/Backload/wiki/Example-12
- add an event handler for the "StoreFileRequestStarted" event
- edit the content type thumbs path in e.Param.FileStatus.Files[0].StorageLocation (if i remember right)

**Comment by j-furby** [comment]
Thanks Auersberg, top notch support. I'm about to leave the office and will pick this back up on Monday. Have a good weekend.

**Comment by Auersberg** [comment]
Sorry the `StoreFileRequestStarted` event will not help, because `e.Param.FileStatusItem.StorageInfo.ContentTypeThumbsPath` is readonly. Just tested.

I asked the developer for this part of code to check it regarding this issue. He reported back, that in some cases a fallback procedure will be triggered is even if this is not necessary. So this issue can be considered as a bug.

This will be fixed in the 2.1 release. I have uploaded a pre-release package to NuGet. You can find it here: https://www.nuget.org/packages/Backload.Core/2.1.0.5-rc2 or install it via Package Manager: 'PM> Install-Package Backload.Core -Pre '.

Thanks for reporting this issue. On Monday I am not in my office but I'll check the emails when I have the time.

**Comment by j-furby** [comment]
Hi Auersberg, 

Sorry for the delay. I've just downloaded the RC and can confirm it is now working on the server. Thanks for all your help!

James.
